crescent/errors.py

```diff
@@ -0,0 +1,78 @@
+from __future__ import annotations
+
+from typing import Any, Awaitable, Callable, Sequence, Type, TypeVar
+
+from crescent.internal.meta_struct import MetaStruct
+from crescent.typedefs import (
+    AutocompleteErrorHandlerCallbackT,
+    CommandErrorHandlerCallbackT,
+    EventErrorHandlerCallbackT,
+)
+
+__all__: Sequence[str] = ("catch_command", "catch_event", "catch_autocomplete")
+
+
+T = TypeVar("T", bound=Callable[..., Awaitable[Any]])
+
+
+def _make_catch_function(
+    error_handler_var: str,
+) -> Callable[[Type[Exception]], Callable[[T | MetaStruct[T, Any]], MetaStruct[T, Any]]]:
+    def func(
+        *exceptions: Type[Exception],
+    ) -> Callable[[T | MetaStruct[T, Any]], MetaStruct[T, Any]]:
+        def decorator(callback: T | MetaStruct[T, Any]) -> MetaStruct[T, Any]:
+            if isinstance(callback, MetaStruct):
+                meta = callback
+            else:
+                meta = MetaStruct(callback, None)
+
+            def app_set_hook(meta: MetaStruct[T, Any]) -> None:
+                for exc in exceptions:
+                    getattr(meta.app, error_handler_var).register(meta, exc)
+
+            meta.app_set_hooks.append(app_set_hook)
+            return meta
+
+        return decorator
+
+    return func
```

Comment:
`app_set_hook` can be put directly under `_make_catch_function`

Reply:
Actually since it uses `exceptions`, `app_set_hook` can go under `func` instead.

Reply:
i just moved it there